Makefile.sp

```diff
@@ -114,7 +114,32 @@ sp-vale: sp-install
 	@echo ""
 	@. $(VENV); vale --config "$(SPHINXDIR)/vale.ini" --glob='*.{md,txt,rst}' $(TARGET)
 
-
+sp-pdf-prep: sp-install
+	@. $(VENV); (dpkg-query -W -f='$${Status}' latexmk 2>/dev/null | grep -c "ok installed" >/dev/null && echo "Package latexmk is installed") || (echo "PDF generation requires the installation of the following packages: latexmk fonts-freefont-otf fonts-ibm-plex texlive-latex-recommended texlive-latex-extra texlive-fonts-recommended texlive-font-utils texlive-lang-cjk texlive-xetex plantuml xindy tex-gyre dvipng" && echo "" && echo "make pdf-prep-force will install these packages" && echo "" && echo "Please be aware these packages will be installed to your system" && false)
```

Comment:
> Is there a way to use a loop on a list of packages here instead of having the full command for each package? 

I pulled this yesterday and modified the Makefile exactly like you're suggesting. I didn't get to submit the suggestion, but I hope to get to finishing it today.

Reply:
I've updated the file to loop through the list. There might be a better way to do it than the way I've implemented though, so feel free to suggest different changes.